fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -1228,27 +1248,49 @@ public void markReadable() throws Exception {
                     return null;
                 }).get();
                 assertThat(recordStore.isIndexReadable(index), is(false));
-
-                indexBuilder.buildRange(recordStore, Key.Evaluated.scalar(1066L), null).get();
-                assertFalse(recordStore.firstUnbuiltRange(index).get().isPresent());
-                assertTrue(recordStore.markIndexReadable(index).get());
-                assertFalse(recordStore.markIndexReadable(index).get());
-                assertThat(recordStore.isIndexReadable(index), is(true));
             }
-            // Not committing.
+        }
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            assertThat(recordStore.isIndexReadable(index), is(false));
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder().setRecordStore(recordStore).setIndex(index)
+                    .build()) {
+                indexBuilder.buildIndex(false);
+            }
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            assertFalse(recordStore.firstUnbuiltRange(index).get().isPresent());
+            assertTrue(recordStore.markIndexReadable(index).get());
+            assertFalse(recordStore.markIndexReadable(index).get());
+            assertThat(recordStore.isIndexReadable(index), is(true));
```

Comment:
It seems like this block should start by doing:
```
assertThat(recordStore.isIndexReadable(index), is(false));
```

Reply:
✅